refactor/reviews

abdo-essam opened the pull request:
# What does this PR do?

This PR refactors the `ReviewsScreen` and its associated components for better organization and clarity.

Key changes include:
- Extracted UI elements into separate composable functions: `ReviewsList`, `ReviewItem`, `ReviewHeader`, `AuthorInfo`, and `EmptyReviewsState`.
- Introduced `ReviewsDimens` to centralize dimension values.
- Updated `ReviewsViewModel` to improve review fetching logic and state management.
- Made `query` parameter in `createPagingSourceFlow` optional.
- Renamed `ReviewsDate` to align with new component structure.
- Corrected parameter name in `GetTvShowUseCase.getTvShowReviews`.

abdo-essam committed:
refactor: Restructure ReviewsScreen and related components

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into refactor/reviews

london-bot[bot] committed:
Branch was auto-updated.

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsViewModel.kt:
```diff
@@ -19,44 +19,57 @@ class ReviewsViewModel @Inject constructor(
+            onStart = {
+                updateState { copy(isLoading = true) }
+            },
             onSuccess = { pagingFlow ->
```
separate the content of onSuccess in a function then call it

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsViewModel.kt:
```diff
@@ -19,44 +19,57 @@ class ReviewsViewModel @Inject constructor(
+                    )
+                }
+            },
+            onError = { errorState ->
```
same

abdo-essam committed:
Refactor(reviews): Extract success/error handlers

abdo-essam committed:
Merge remote-tracking branch 'origin/refactor/reviews' into refactor/reviews

Abdulrahman-Ragab-01 reviewed (dismissed):
Great Job

london-bot[bot] committed:
Branch was auto-updated.

marwanqashwa commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,101 @@
+                onBackClick = reviewContract::onBackClicked
+            )
+
+            when {
```
i think no need for when  it's two branches only

abdo-essam committed:
Refactor: Reviews screen and add ReviewsViewModel tests

This commit refactors the ReviewsScreen for clarity and adds comprehensive unit tests for the ReviewsViewModel to ensure its logic is robust and reliable.

london-bot[bot] committed:
Branch was auto-updated.

london-bot[bot] committed:
Branch was auto-updated.

abdo-essam committed:
refactor: Consolidate ReviewsScreen components

Moves all composables from `presentation/src/main/java/com/london/presentation/feature/reviews/components` into `ReviewsScreen.kt`.

BassantM commented on domain/src/main/java/com/london/domain/usecase/details/tvshow/GetTvShowUseCase.kt:
```diff
@@ -54,11 +54,11 @@ class GetTvShowUseCase @Inject constructor(
-    suspend fun getTvShowReviews(movieId: Int, pageNumber: Int): PagedFetchResponse<ReviewEntity> =
-        tvShowRepository.getTvShowReviews(movieId, pageNumber)
+    suspend fun getTvShowReviews(tvShowId: Int, pageNumber: Int): PagedFetchResponse<ReviewEntity> =
+        tvShowRepository.getTvShowReviews(tvShowId, pageNumber)
```
add named argument

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+            is ReviewEffect.NavigateBack -> onNavigateBack()
+        }
+    }
+
```
Add a build screen that contains content composable

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+) {
+    val reviewsList = uiState.reviews.collectAsLazyPagingItems()
+
+    BuildScreen(
```
extract this

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+                ReviewsList(
+                    reviewsList = reviewsList,
+                    modifier = Modifier.fillMaxSize()
+                )
```
extract this to be a private function

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+}
+
+@Composable
+fun EmptyReviewsState(
```
make it private function

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+}
+
+@Composable
+fun ReviewsList(
```
same private function

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+}
+
+@Composable
+fun ReviewItem(
```
same

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+}
+
+@Composable
+fun ReviewHeader(
```
same

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+}
+
+@Composable
+fun AuthorInfo(
```
same

BassantM commented on presentation/src/main/java/com/london/presentation/feature/reviews/ReviewsScreen.kt:
```diff
@@ -0,0 +1,334 @@
+}
+
+@Composable
+fun ReviewsDate(
```
same

abdo-essam committed:
refactor: reviews screen composable functions

Make composable functions private.
Pass named parameter to getTvShowReviews

abdo-essam committed:
Refactor(reviews): Move reviews list logic to composable

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into refactor/reviews

# Conflicts:
#	domain/src/main/java/com/london/domain/usecase/details/tvshow/GetTvShowUseCase.kt

abdo-essam committed:
refactor(domain): Rename id to tvShowId in getTvShowReviews

london-bot[bot] committed:
Branch was auto-updated.
